contrib/comdb2makecluster

```diff
@@ -0,0 +1,251 @@
+    usage
+fi
+
+
+TMPDIR=${TMPDIR:-/tmp}
+CDB2_CONFIG=$DBDIR/comdb2db.cfg
+CDB2_OPTIONS="--cdb2cfg ${CDB2_CONFIG}"
+COMDB2_EXE=comdb2
+COMDB2AR_EXE=comdb2ar
+export comdb2ar=$COMDB2AR_EXE  # used by copycomdb2
+export COMDB2AR_EXOPTS="-x $COMDB2_EXE"
+export COMDB2_ROOT=${DBDIR}
+myhostname=`hostname`
+
+check_executables_for_cluster() 
+{
+    for node in $CLUSTER; do
+        if [ $node == $myhostname ] ; then
+            readlink -f $DBDIR && failexit "$DBDIR already exists, please clean up before attempting to start"
+            which $COMDB2_EXE > /dev/null || failexit "$COMDB2_EXE not in path"
+            which copycomdb2  > /dev/null || failexit "copycomdb2 not in path"
+            continue
+        fi
+
+        ssh -o StrictHostKeyChecking=no $node "ls $DBDIR &> /dev/null" < /dev/null && failexit "$DBDIR already exists in node $node, please clean up before attempting to start"
+        ssh -o StrictHostKeyChecking=no $node "which $COMDB2_EXE > /dev/null" < /dev/null || failexit "$COMDB2_EXE not in path in node $node"
+        ssh -o StrictHostKeyChecking=no $node "which $COMDB2AR_EXE > /dev/null" < /dev/null || failexit "$COMDB2AR_EXE not in path in node $node"
+    done
+}
+
+check_executables_for_cluster
+
+
+mkdir -p $DBDIR $TMPDIR
+
+# setup files:
+echo "$DBNAME: creating"
+LRL="$DBDIR/$DBNAME.lrl"
+> ${LRL}
+
+if [[ -n $USESSL ]] ; then
+    cat >> $DBDIR/$DBNAME.lrl <<EOPTIONS
+ssl_client_mode REQUIRE
+ssl_cert_path $DBDIR
+EOPTIONS
+fi
+
+cat >> $DBDIR/$DBNAME.lrl <<EOPTIONS
+name    $DBNAME
+dir     $DBDIR
+
+EOPTIONS
+
+df $DBDIR | awk '{print $1 }' | grep "tmpfs\|nfs" > /dev/null && echo "setattr directio 0" >> ${LRL}
+
+echo $DBNAME 0 $CLUSTER > $CDB2_CONFIG
+echo "comdb2_config:default_type=testsuite" >> $CDB2_CONFIG
+echo "cluster nodes $CLUSTER" >> $DBDIR/$DBNAME.lrl
+
+for node in $CLUSTER; do #verify we can ssh to nodes
+    if [ $node == $myhostname ] ; then
+        continue
+    fi
+    hst=$(ssh -o StrictHostKeyChecking=no $node "hostname")
+    if [ "x$hst" == "x" ] ; then
+        failexit "cant get hostname on $node"
+    fi
+done
+
+# Configure client SSL
+echo "comdb2_config:ssl_cert_path=$DBDIR" >>$CDB2_CONFIG
+
+set +e
+
+pmux_port=5105
+pmux_cmd='pmux -n'
+if [ -n "$PMUXPORT" ] ; then
+    pmux_port=$PMUXPORT
+    pmux_socket=/tmp/pmux.socket.$PMUXPORT
+    pmux_port_range="-r 21000:22000"
+    pmux_cmd="pmux -n -p $PMUXPORT -b $pmux_socket $pmux_port_range"
+    echo "comdb2_config:portmuxport=$PMUXPORT" >> $CDB2_CONFIG
+    echo "portmux_port $PMUXPORT" >> ${LRL}
+    echo "portmux_bind_path $pmux_socket" >> ${LRL}
+fi
+stop_pmux="pgrep pmux > /dev/null && (exec 3<>/dev/tcp/localhost/${pmux_port} && echo exit >&3 )"
+
+
+
+
+keygen() {
+# Setup ssl certificate
+    openssl req -x509 -newkey rsa:4096 -keyout ${DBDIR}/server.key \
+        -out ${DBDIR}/server.crt \
+        -days 365 -nodes -subj \
+        "/C=US/ST=New York/L=New York/O=Bloomberg/OU=Comdb2/CN=*.bloomberg.com" &>> $DBDIR/setup.log
```

Comment:
I don't think we should generate SSL keypairs for end-users.

Reply:
Ok. Was not sure about ssl keys. Easy to remove.